src/com/invoca/docker/Image.groovy

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/groovy
+package com.invoca.docker
+
+class Image implements Serializable {
+  public static String LABEL_SCHEMA_VERSION = "org.label-schema.schema-version"
+  public static String LABEL_BUILD_DATE = "org.label-schema.build-date"
+  public static String LABEL_VCS_URL = "org.label-schema.vcs-url"
+
+  private String imageName
+  private String[] tags
+  private String baseDir
+  private Script script
+  private String[] imageNameWithTags
+  private String[] labels
+
+  public Image(Script script, String imageName, String tag, String baseDir = ".") {
+    this(script, imageName, [tag], baseDir)
+  }
+
+  public Image(Script script, String imageName, ArrayList tags, String baseDir = ".") {
+    this(script, imageName, tags as String[], baseDir)
+  }
+
+  public Image(Script script, String imageName, String[] tags, String baseDir = ".") {
+    this.script = script
+    this.imageName = imageName
+    this.tags = tags
+    this.baseDir = baseDir
```

Comment:
Could you explain why this one is different and isn't written like the others?

```groovy
this(script, imageName, tags, baseDir)
```

Reply:
A `this()` method call from inside a constructor is calling another constructor on the same class, but with a different signature.

Java (and Groovy) supports method overloading, where you can have multiple methods with the same name, as long as their method signatures are different.  The first two constructors are essentially just adapters.  They take the argument list and alter it to fit the signature of the 3rd constructor, which performs the assignment.

The first constructor takes a single tag and converts it to an array, which matches the 3rd signature.  The second coerces an `ArrayList` of tags into a string array, which also matches the 3rd signature.